cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -1567,6 +1588,24 @@ const FunctionDoc log1p_checked_doc{
      "Use function \"log1p\" if you want non-positive values to return "
      "-inf or NaN."),
     {"x"}};
+
+const FunctionDoc floor_doc{
+    "Calculate the greatest integer in magnitude less than or equal to the "
+    "argument element-wise",
+    "",
```

Comment:
```suggestion
    "Round down to the nearest integer",
    "Calculate the greatest integer in magnitude less than or equal to the "
    "argument element-wise",
```

(that's how you explained it in the api_scalar.h  doc comments, which I find easier to understand as short summary of the function.